framework/sysdeps/unix/child_debug.cpp

```diff
@@ -1036,16 +1045,25 @@ void debug_crashed_child()
 
         if (ctx.contents == CrashContext::NoContents)
             break;
+
+        // find out which slice this context is for
+        int slice = find_slice_by_pid(ctx.fixed.pid, children);
+        if (slice < 0) {
+            log_message(-1, "Received a context dump from PID %ld that doesn't seem to match any "
```

Comment:
We need to have a log for internal errors. I've seen quite a few places where we cannot actually continue, but it's not logged as our own problem.

Reply:
This here is expected to be an impossible condition (I could have used an `assert`). The socket is created in 
https://github.com/opendcdiag/opendcdiag/blob/dbb1bae6e9a41973af13e8728a4aed17303602d7/framework/sysdeps/unix/child_debug.cpp#L374-L386
And as a socketpair that is not accessible to any other processes, at least not accidentally. So the only thing that should write a PID is the OpenDCDiag framework itself.

The other way of doing this would be to use [`SCM_CREDENTIALS`](https://man7.org/linux/man-pages/man7/unix.7.html), but since users often run the tool as root, a corrupt child process would still be able to send corrupt data.